fish-rust/src/builtins/ulimit.rs

```diff
@@ -1,16 +1,468 @@
+use std::cmp::Ordering;
+
+use libc::{c_uint, rlim_t, RLIM_INFINITY};
+use nix::errno::Errno;
+use once_cell::sync::Lazy;
+
+use crate::{compat::*, fallback::*};
+
 use super::prelude::*;
 
+fn getrlimit(resource: c_uint) -> Result<(rlim_t, rlim_t), Errno> {
+    use nix::sys::resource;
+    let resource: i32 = resource.try_into().unwrap();
+    resource::getrlimit(unsafe { std::mem::transmute(resource) }) // Resource is #[repr(i32)] so this is ok
+}
+
+fn setrlimit(resource: c_uint, rlim_cur: rlim_t, rlim_max: rlim_t) -> Result<(), Errno> {
+    use nix::sys::resource;
+    let resource: i32 = resource.try_into().unwrap();
+    resource::setrlimit(
+        unsafe { std::mem::transmute(resource) }, // Resource is #[repr(i32)] so this is ok
+        rlim_cur,
+        rlim_max,
+    )
+}
+
+/// Print the value of the specified resource limit.
+fn print(resource: c_uint, hard: bool, streams: &mut IoStreams) {
+    let l = get(resource, hard);
+
+    if l == RLIM_INFINITY {
+        streams.out.append(wgettext_fmt!("unlimited\n"));
```

Comment:
We have momentarily banned wgettext_fmt without format args

```suggestion
        streams.out.append(wgettext!("unlimited\n"));
```